tests/unit/source_code/notebooks/test_sources.py

```diff
@@ -21,6 +23,36 @@ def test_file_linter_lints_not_yet_supported_language(path, migration_index):
     assert [advice.code for advice in advices] == ["unsupported-language"]
 
 
+class FriendFileLinter(FileLinter):
```

Comment:
Why `Friend`?

Reply:
We have linting rules that block access to protected members. It's normal (and conventional) in most languages for tests to have better-than-public access to the internals of the thing being tested: unit tests are often (intentionally) tightly coupled to the thing under test and are necessarily aware of the implementation and internals.

In C++ there's a concept of "friend" classes: these are allowed to access protected and private attributes. (The package-private concept of Java is similar, and a major reason that unit tests are conventionally placed in the same package as the classes being tested.)

So tying things together: `Friend` is a hint that it's purpose is to access protected attributes and allow them to be used during testing.

Reply:
Thanks for the explanation! Learned something new here :)